rkt/enter.go

```diff
@@ -81,18 +94,35 @@ func runEnter(args []string) (exit int) {
 		return 1
 	}
 
-	if _, err = os.Stat(filepath.Join(common.AppRootfsPath(c.path(), *imageID))); err != nil {
-		stderr("Unable to access app rootfs: %v", err)
+	argv, err := getEnterArgv(c, imageID, args)
+	if err != nil {
+		stderr("Enter failed: %v", err)
 		return 1
 	}
 
-	argv, err := getEnterArgv(c, imageID, args)
+	ds, err := cas.NewStore(globalFlags.Dir)
 	if err != nil {
-		stderr("Enter failed: %v", err)
+		stderr("Cannot open store: %v", err)
 		return 1
 	}
 
-	if err = stage0.Enter(c.path(), imageID, argv); err != nil {
+	stage1ID, err := c.getStage1Hash()
+	if err != nil {
+		stderr("Error getting stage1 hash")
+		return 1
+	}
+
+	enterPath := ""
+	useOverlay, err := runningWithOverlay(c.path())
+	if err != nil {
+		stderr("Error determining if the container uses overlay: %v")
+	}
+	if useOverlay {
```

Comment:
Yeah, I had it the way you describe before but I was concerned about the tree cache changing. Now that I think of if, it doesn't make sense to make the distinction: if the tree cache changes it would be under a different hash so assuming the user doesn't remove it manually it should be fine.

I'll change it back.